benchmark-scripts/collect_platform_metrics.sh

```diff
@@ -35,112 +35,96 @@ then
-    pcm_pids=($(ps aux | grep pcm | grep -v grep | awk '{print $2}'))
-    if [ -z "$pcm_pids" ]
-    then
-      echo "no dangling pcm background processes to clean up"
-    else
-      for pid in "${pcm_pids[@]}"
-      do
-        echo "cleaning up dangling pcm $pid"
-        sudo kill -9 "$pid"
-      done
-    fi
-    timeout "$DURATION" sudo $PCM_DIRECTORY/pcm 1 -silent -nc -nsys -csv=$LOG_DIRECTORY/pcm.csv &
+    timeout "$DURATION" $PCM_DIRECTORY/pcm 1 -silent -nc -nsys -csv=$LOG_DIRECTORY/pcm.csv &
     echo "DEBUG: pcm started collecting"
   fi
       
   # DGPU pipeline and Flex GPU Metrics
   if [ "$PLATFORM" == "dgpu" ] && [ $HAS_ARC == 0 ] 
   then
+    cpuOutputDir=/tmp/xpumdump/
     metrics=0,5,22,24,25
+    device=0
     # Check for up to 4 GPUs e.g. 300W max 
     if [ -e /dev/dri/renderD128 ]; then
-      echo "==== Starting xpumanager capture (gpu 0) ===="
-      timeout "$DURATION" sudo xpumcli dump --rawdata --start -d 0 -m $metrics -j > ${LOG_DIRECTORY}/xpum0.json &
+      device=0
+      echo "==== Found device $device ===="
     fi
     if [ -e /dev/dri/renderD129 ]; then
-      echo "==== Starting xpumanager capture (gpu 1) ===="
-      timeout "$DURATION" sudo xpumcli dump --rawdata --start -d 1 -m $metrics -j > ${LOG_DIRECTORY}/xpum1.json &
+      device=1
+      echo "==== Found device $device ===="
     fi
     if [ -e /dev/dri/renderD130 ]; then
-      echo "==== Starting xpumanager capture (gpu 2) ===="
-      timeout "$DURATION" sudo xpumcli dump --rawdata --start -d 2 -m $metrics -j > ${LOG_DIRECTORY}/xpum2.json &
+      device=2
+      echo "==== Found device $device ===="
     fi
     if [ -e /dev/dri/renderD131 ]; then
-      echo "==== Starting xpumanager capture (gpu 4) ===="
-      timeout "$DURATION" sudo xpumcli dump --rawdata --start -d 3 -m $metrics -j > ${LOG_DIRECTORY}/xpum3.json &
+      device=3
+      echo "==== Found device $device ===="
     fi
+    echo "==== Starting xpumanager capture (gpu $device) ===="
+    docker run -itd -v $SOURCE_DIR/$LOG_DIRECTORY:/$cpuOutputDir  --cap-drop ALL --cap-add CAP_SYS_ADMIN --user root -e XPUM_REST_NO_TLS=1 -e XPUM_EXPORTER_NO_AUTH=1 -e XPUM_EXPORTER_ONLY=1 --publish 127.0.0.1:29999:29999 --device /dev/dri:/dev/dri --name=xpum$device intel/xpumanager:v1.0.0 
+    sleep 10
+    docker exec xpum$device bash -c "xpumcli dump --rawdata --start -d 0 -m $metrics -j"
   # DGPU pipeline and  Arc GPU Metrics
   elif [ "$PLATFORM" == "dgpu" ] && [ $HAS_ARC == 1 ]
   then
     echo "==== Starting igt arc ===="
     # Arc is always on Core platform and although its GPU.1, the IGT device is actually 0
     # Collecting both 
-    timeout $DURATION ../docker-run-igt.sh 0
-    timeout $DURATION ../docker-run-igt.sh 1
+    timeout $DURATION docker run -v $SOURCE_DIR/$LOG_DIRECTORY:/$LOG_DIRECTORY -e LOG_DIRECTORY=$LOG_DIRECTORY -itd --privileged benchmark:igt bash -c "/usr/local/bin/intel_gpu_top -d pci:card=0 -J > $LOG_DIRECTORY/igt0.json"
```

Comment:
Does it make sense to still have a different container for igt? should we just include it in this one? Maybe in the future.

Reply:
For now I think it makes sense to keep them separate but in the future we can possibly merge if the same GPU tool is used for core and dgpu